recipes/lcms/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+    topics = ("conan", "lcms", "cmm", "icc", "cmm-engine")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, 'fPIC': True}
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename('Little-CMS-lcms%s' % self.version, self._source_subfolder)
+
+    def _build_visual_studio(self):
+        # since VS2015 vsnprintf is built-in
+        if int(str(self.settings.compiler.version)) >= 14:
+            path = os.path.join(self._source_subfolder, 'src', 'lcms2_internal.h')
+            tools.replace_in_file(path, '#       define vsnprintf  _vsnprintf', '')
+
+        with tools.chdir(os.path.join(self._source_subfolder, 'Projects', 'VC2013')):
+            target = 'lcms2_DLL' if self.options.shared else 'lcms2_static'
+            upgrade_project = True if int(str(self.settings.compiler.version)) > 12 else False
+            # run build
+            msbuild = MSBuild(self)
+            msbuild.build("lcms2.sln", targets=[target], platforms={"x86": "Win32"}, upgrade_project=upgrade_project)
+
+    def _build_configure(self):
+        env_build = AutoToolsBuildEnvironment(self)
+        with tools.chdir(self._source_subfolder):
+            args = ['prefix=%s' % self.package_folder]
+            if self.options.shared:
+                args.extend(['--disable-static', '--enable-shared'])
+            else:
+                args.extend(['--disable-shared', '--enable-static'])
+            env_build.configure(args=args)
+            env_build.make()
+            env_build.make(args=['install'])
+
+    def build(self):
+        if self.settings.compiler == 'Visual Studio':
+            self._build_visual_studio()
+        else:
+            self._build_configure()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        if self.settings.compiler == 'Visual Studio':
+            self.copy(pattern='*.h', src=os.path.join(self._source_subfolder, 'include'), dst='include', keep_path=True)
+            if self.options.shared:
+                self.copy(pattern='*.lib', src=os.path.join(self._source_subfolder, 'bin'), dst='lib', keep_path=False)
+                self.copy(pattern='*.dll', src=os.path.join(self._source_subfolder, 'bin'), dst='bin', keep_path=False)
+            else:
+                self.copy(pattern='*.lib', src=os.path.join(self._source_subfolder, 'Lib', 'MS'), dst='lib',
+                          keep_path=False)
+        # remove entire share directory
+        shutil.rmtree(os.path.join(self.package_folder, 'share'), ignore_errors=True)
```

Comment:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, 'share'))
```

Reply:
`rmtree` is here because of Windows build that do not package `share` directory

Reply:
I know, the point is using Conans.tools instead. It has the same effect, but just trying to keep a pattern on the recipes.

Reply:
Ah, got it! I confused it with `os.rmdir`